src/globus_sdk/experimental/globus_app/globus_app.py

```diff
@@ -160,10 +172,18 @@ def __init__(
         self.app_name = app_name
         self.config = config
 
-        if login_client and (client_id or client_secret):
-            raise GlobusSDKUsageError(
-                "login_client is mutually exclusive with client_id and client_secret."
-            )
+        if login_client:
+            if client_id or client_secret:
+                raise GlobusSDKUsageError(
+                    "login_client is mutually exclusive with client_id and "
+                    "client_secret."
+                )
+            if login_client.environment != self.config.environment:
+                raise GlobusSDKUsageError(
+                    f"[Environment Mismatch] The login_client's environment "
```

Comment:
Did flake8 (or flake8-bugbear, or whatever) ignore this f-string with no variables in it? I prefer the consistency but thought there was a linter that would object to this.